target/linux/lantiq/image/Makefile

```diff
@@ -130,6 +130,11 @@ include tp-link.mk
 include vr9.mk
 endif
 
+ifeq ($(SUBTARGET),xrx300)
+DEFAULT_SOC := grx390
+include grx390.mk
+endif
```

Comment:
Do we really need another subtarget here?

Reply:
> Do we really need another subtarget here?

Lantiq SDK split xrx200 and xrx300 into separate subtargets.
Differences:
- newer builtin WiFi (WAVE400 vs WAVE300)
- gphy in ar10 switch use another firmware
- newer vdsl (VRX318)
- 3x PCIe vs 1xPCI+1xPCIe (OpenWRT PCIe driver didn't work with grx390 and I have plan to use mainline dw-pcie driver)

At this point, I can merge in with xrx200. There may be a problem with this in the future.

Reply:
* The xrx200 does not have any wifi integrated, I think a wave300 is integrated in the xrx300/ xrx330 family, wave400 does not exist (unlucky number)
* The VRX318 is an external DSL PCIe chip
* The NAND core is the same as in the xrx200 just some more extensions. 

The GRX350 / 550 is a different chip with a InterAptiv CPU.

Here is a recent NAND driver:
https://lkml.org/lkml/2020/6/16/302
It is the same NAND controller as in the xrx300, you just have an older version, it could be that it does not support all features.

A PCIe controller driver is also upstream:
https://git.kernel.org/linus/ed22aaaede44f647477a5048e62855c0ed49c9bd
I am not sure if the xrx300 uses a Lantiq PCIe PHY like the xrx200 or a Synopsys PCIe PHY like the xrx500 and LGM. If it uses a synopsys PCIe PHY it should be easy to make this driver work.
The xrx200 PCI (without express) controller driver is stil using the old MIPS PCIe driver framework which is a problem.

Reply:
Thanks
I didn't know that they are almost identical. I moved it to the xrx200 subtarget.